osquery/tables/applications/linux/namespace_ops.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <fcntl.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include <osquery/tables/applications/linux/namespace_ops.h>
+
+namespace osquery {
+namespace tables {
+
+Status NamespaceOps::invoke(std::function<void(int fd)> fn) {
+  std::string pidns = "/proc/" + std::to_string(_pid) + "/ns/mnt";
```

Comment:
Make use of the direct initialization over copy: `std::string pidns ("/proc/" + std::to_string(_pid) + "/ns/mnt");`